ProcessLib/BoundaryCondition/BoundaryConditionCollection.cpp

```diff
@@ -52,6 +38,14 @@ void BoundaryConditionCollection::addBCsForProcessVariables(
                   std::back_inserter(_boundary_conditions));
     }
 
-    initializeDirichletBCs(_boundary_conditions, _dirichlet_bcs);
+    // for each BC there will be storage for Dirichlet BC, which by default is
+    // empty and has to be filled by the respective BC object if needed.
+    _dirichlet_bcs.resize(_boundary_conditions.size());
```

Comment:
Each BC can be either Dirichlet (or "essential") or natural, so each of the BCs need some storage for it. As you wrote, this implies that `_dirichlet_bcs` is not empty. However, the memory overhead for that should be rather small, because only some empty `std::vector`s are created.